packages/rest/src/parser.ts

```diff
@@ -123,6 +126,28 @@ function buildOperationArguments(
         );
     }
   }
-  if (requestBodyIndex > -1) paramArgs.splice(requestBodyIndex, 0, body);
-  return paramArgs;
+
+  debug('Coercing parameters', paramArgs);
+
+  const coercedParamArgs: OperationArgs = [];
+  const paramObjects = operationSpec.parameters;
+  // coercion is done IFF parameters are defined in the OpenAPI spec
+  if (paramObjects) {
+    for (let i = 0; i < paramArgs.length; i++) {
+      const paramObject = paramObjects[i];
+      if (!isReferenceObject(paramObject)) {
+        // is a ParameterObject
+        if (paramObject.schema) {
+          if (isSchemaObject(paramObject.schema)) {
+            // basic type
+            const serializer = getSerializer(paramObject.schema.type!);
+            coercedParamArgs.push(serializer.coerce(paramArgs[i]));
```

Comment:
I have mixed feelings about rolling out our own coercion implementation. At one hand, it allows us to use the same coercion algorithm in multiple places (e.g. REST, ORM/database, etc.) and have full control over the coercion rules. At the other hand, there are many edge cases that are tricky to get right and we can easy end up with a behavior that's surprising for users because it's inconsistent with other coercion implementations (LB 3.x, [AJV coercion](https://www.npmjs.com/package/ajv#coercing-data-types)).

Reply:
I feel coercion rules from `string` to whatever seem consistent for most coercion implementations. I like your suggestion in your main comment down below and I think it will make for a good starting point.

Reply:
In this context, it's probably better to talk about `deserialization`. The responsibility is to extract `typed/formatted` values from http requests (a set of strings from path/query/header/body/...).